Tasks/Maven/maventask.ts

```diff
@@ -284,12 +287,28 @@ function isCodeAnalysisUploadRequired():boolean {
     return tl.getInput('pmdAnalysisEnabled', true) == 'true';
 }
 
-// Take a result object and attempt to upload the XML and HTML reports as artifacts. Throws if fail.
+function cleanDirectory(targetDirectory:string):Q.Promise<void> {
+    var defer = Q.defer<void>();
```

Comment:
As there is still a limited possibility of the delete operation being delayed, the cleanDirectory function uses the Q promise framework (see other uses of the same, i.e. in the toolrunner) to enforce synchronous behaviour.